### PR TITLE
Added scope document and made changes to pull request template

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
## Description

Based on the conversation held by the expansion Senate, this is the scope document that was agreed upon as deciding "what is in scope". This PR adds that documents and links to it in the PR template.

## **People who collaborated with me in this PR**
@Pawkkie and arguably the entire senate

## **Discord contact info**
psf

thread: https://discord.com/channels/419213663107416084/1309297830648745996

### CODE REVIEW

File: docs/scope.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* **Showdown Supported (SS)**: A core series game game which metagame can be played on Showdown.  
+  * Notably, this is every [core series game](https://bulbapedia.bulbagarden.net/wiki/Core_series#List_of_core_series_games) except Pokémon Legends: Arceus and Pokémon Legends: Z-A.  
+* **Base Expansion Version**: A .gba file built from an unmodified `master` branch of `pokeemerald-expansion`.  
+* **Vanilla Emerald Version**: A .gba file built from an unmodified `master` branch of `pokeemerald`.
+
+# Guidelines
+
+A pull request meets the scope crtieria if:
+* The feature does not belong to a category considered “not in scope” AND
+* The feature belongs to a category conisred “in scope”
```
Comment (Pawkkie) [inline docs/scope.md:16]:
```suggestion
* The feature belongs to a category considered “in scope”
```

File: docs/scope.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## In Scope Categories
+
+1. **SS Species:** Adds Species that have appeared in a Showdown-supported title  
+2. **SS Moves:** Adds Moves and Move Animations that have appeared in a Showdown-supported title  
+3. **SS Abilities:** Adds Abilities that have appeared in a Showdown-supported title  
+4. **SS Items:** Adds Items that have appeared in a Showdown-supported title  
+5. **SS Gimmicks:** Adds Gimmicks that have appeared in a Showdown-supported title  
+6. **SS Battle Types:** Adds Special Battle Types that have appeared in a Showdown-supported title  
+7. **SS Battle Mechanics:** Add mechanical battle changes that have appeared in a Showdown-supported title
```
Comment (Pawkkie) [inline docs/scope.md:26]:
```suggestion
7. **SS Battle Mechanics:** Adds mechanical battle changes that have appeared in a Showdown-supported title
```

File: docs/scope.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+4. **SS Items:** Adds Items that have appeared in a Showdown-supported title  
+5. **SS Gimmicks:** Adds Gimmicks that have appeared in a Showdown-supported title  
+6. **SS Battle Types:** Adds Special Battle Types that have appeared in a Showdown-supported title  
+7. **SS Battle Mechanics:** Add mechanical battle changes that have appeared in a Showdown-supported title  
+8. **Improve Battle AI:** Improve the Battle AI in a way that allows it to approach the skill and capability of a human competitive player  
+9. **Base Link Compatibility:** Link compatibility with base  
+10. **SS Overworld / Menu Updates:** Replicate overworld or menu changes from Showdown-supported Pokémon titles  
+11. **Speed Up:** Speed up the player experience of features found in base  
+12. **Compression:** Automatically compress assets  
+13. **Novel Experience:** Add a novel experience included in another Showdown Supported title
```
Comment (Pawkkie) [inline docs/scope.md:32]:
```suggestion
13. **Novel Experience:** Adds a novel experience included in another Showdown Supported title
```

File: docs/scope.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Not In Scope Categories
+
+1. **Non-SS Species**: Adds Species that have NOT appeared in a Showdown-supported title  
+2. **Non-SS Moves**: Adds Moves and Move Animations that have NOT appeared in a Showdown-supported title  
+3. **Non-SS Abilities**: Adds Abilities that have NOT appeared in a Showdown-supported title  
+4. **Non-SS Items**: Adds Items that have NOT appeared in a Showdown-supported title  
+5. **Non-SS Gimmicks**: Adds Gimmicks that have NOT appeared in a Showdown-supported title  
+6. **Non-SS Battle Types**: Adds Special Battle Types that have NOT appeared in a Showdown-supported title  
+7. **Duplicate Feature UI**: Add functionality that duplicates the core functionality of an existing vanilla feature
```
Comment (Pawkkie) [inline docs/scope.md:43]:
```suggestion
7. **Duplicate Feature UI**: Adds functionality that duplicates the core functionality of an existing vanilla feature
```

File: docs/scope.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+4. **Non-SS Items**: Adds Items that have NOT appeared in a Showdown-supported title  
+5. **Non-SS Gimmicks**: Adds Gimmicks that have NOT appeared in a Showdown-supported title  
+6. **Non-SS Battle Types**: Adds Special Battle Types that have NOT appeared in a Showdown-supported title  
+7. **Duplicate Feature UI**: Add functionality that duplicates the core functionality of an existing vanilla feature  
+8. **Vanilla Link Compatibility**: Link compatibility with vanilla  
+9. **External Program**: External programs
+
+## Discussion Required Categories
+
+Pull Requests that fall into this category should be brought up to maintainers, who will discuss and vote as to whether or not the feature is considered in scope. Considerations for acceptance may include invasiveness of implementation, popularity, ease of maintanence, etc.
```
Comment (Pawkkie) [inline docs/scope.md:49]:
```suggestion
Pull Requests that fall into this category should be brought up to maintainers, who will discuss and vote as to whether or not the feature is considered in scope. Considerations for acceptance may include invasiveness of implementation, popularity, ease of maintenance, etc.
```

**Review by Pawkkie — COMMENTED** [review]
Minor typos / grammar / parallel structure we missed in initial reading

File: docs/scope.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Document Purpose
+
+This document is a guide for contributors and Senate to decide if a feature is within "scope" for pokeemerald-expansion. If a feature is not in scope, then it should not be merged. Even if an opened PR is within scope, this does not mean it will be merged, as acceptance criteria will often come down to the details of the implementation.
+
+# Definitions
+
+* **Showdown Supported (SS)**: A core series game who's metagame can be played on Showdown.
+  * Notably, this is every [core series game](https://bulbapedia.bulbagarden.net/wiki/Core_series#List_of_core_series_games) except Pokémon Legends: Arceus and Pokémon Legends: Z-A.  
+* **Base Expansion Version**: A .gba file built from an unmodified `master` branch of `pokeemerald-expansion`.  
+* **Vanilla Emerald Version**: A .gba file built from an unmodified `master` branch of `pokeemerald`.
```
Comment (AsparagusEduardo) [inline docs/scope.md:10]:
```suggestion
* **Vanilla Emerald Version**: A .gba file built from an unmodified `master` branch of pret's `pokeemerald`.
```

**Review by AsparagusEduardo — COMMENTED** [review]
Small adjustment. Otherwise, lgtm :(

File: docs/scope.md (hunk 6 above)
Comment (AsparagusEduardo) [reply docs/scope.md]: Keyboard did :( instead of :), sorry 😅

File: docs/scope.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+# Definitions
+
+* **Showdown Supported (SS)**: A core series game who's metagame can be played on Showdown.
+  * Notably, this is every [core series game](https://bulbapedia.bulbagarden.net/wiki/Core_series#List_of_core_series_games) except Pokémon Legends: Arceus and Pokémon Legends: Z-A.  
+* **Base Expansion Version**: A .gba file built from an unmodified `master` branch of `pokeemerald-expansion`.  
+* **Vanilla Emerald Version**: A .gba file built from an unmodified `master` branch of pret's `pokeemerald`.
+
+# Guidelines
+
+A pull request meets the scope crtieria if:
```
Comment (Bassoonian) [inline docs/scope.md:14]:
```suggestion
A pull request meets the scope criteria if:
```

**Review by hedara90 — DISMISSED** [review]

File: docs/scope.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+# Document Purpose
+
+This document is a guide for contributors and Senate to decide if a feature is within "scope" for pokeemerald-expansion. If a feature is not in scope, then it should not be merged. Even if an opened PR is within scope, this does not mean it will be merged, as acceptance criteria will often come down to the details of the implementation.
+
+# Definitions
+
+* **Showdown Supported (SS)**: A core series game who's metagame can be played on Showdown.
+  * Notably, this is every [core series game](https://bulbapedia.bulbagarden.net/wiki/Core_series#List_of_core_series_games) except Pokémon Legends: Arceus and Pokémon Legends: Z-A.
```
Comment (AsparagusEduardo) [inline docs/scope.md:8]: We don't know yet how Legends Z-A will play or if it'll be supported by Showdown.

File: docs/scope.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Document Purpose
+
+This document is a guide for contributors and Senate to decide if a feature is within "scope" for pokeemerald-expansion. If a feature is not in scope, then it should not be merged. Even if an opened PR is within scope, this does not mean it will be merged, as acceptance criteria will often come down to the details of the implementation.
+
+# Definitions
+
+* **Showdown Supported (SS)**: A core series game who's metagame can be played on Showdown.
+  * Notably, this is every [core series game](https://bulbapedia.bulbagarden.net/wiki/Core_series#List_of_core_series_games) except Pokémon Legends: Arceus and Pokémon Legends: Z-A.  
+* **Base Expansion Version**: A .gba file built from an unmodified `master` branch of `pokeemerald-expansion`.
```
Comment (AsparagusEduardo) [inline docs/scope.md:9]: Should we add a description for upcoming or tagged releases?

File: docs/scope.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+1. **Non-SS Species**: Adds Species that have NOT appeared in a Showdown-supported title  
+2. **Non-SS Moves**: Adds Moves and Move Animations that have NOT appeared in a Showdown-supported title  
+3. **Non-SS Abilities**: Adds Abilities that have NOT appeared in a Showdown-supported title  
+4. **Non-SS Items**: Adds Items that have NOT appeared in a Showdown-supported title  
+5. **Non-SS Gimmicks**: Adds Gimmicks that have NOT appeared in a Showdown-supported title  
+6. **Non-SS Battle Types**: Adds Special Battle Types that have NOT appeared in a Showdown-supported title  
+7. **Duplicate Feature UI**: Adds functionality that duplicates the core functionality of an existing vanilla feature
+8. **Vanilla Link Compatibility**: Link compatibility with vanilla  
+9. **External Program**: External programs
```
Comment (AsparagusEduardo) [inline docs/scope.md:45]: We should probably specify that we mean external programs by those that don't automatically work as part of the `make` process.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: docs/scope.md (hunk 9 above)
Comment (pkmnsnfrn) [reply docs/scope.md:9]: I'm not sure what you mean, could you please elaborate?

File: docs/scope.md (hunk 10 above)
Comment (pkmnsnfrn) [reply docs/scope.md:45]: Agreed, will add once https://github.com/rh-hideout/pokeemerald-expansion/pull/5706#discussion_r1866265003 is addressed

File: docs/scope.md (hunk 9 above)
Comment (pkmnsnfrn) [reply docs/scope.md:9]: Oh I think I understand now: I don't think so, but if you suggest some, I won't fight you

Comment (AsparagusEduardo) [reply docs/scope.md:9]: Based on the description of "A .gba file built from an unmodified `master` branch of `pokeemerald-expansion`.", it made me ask "then what would be a gba file based off upcoming?

Comment (AsparagusEduardo) [reply docs/scope.md:9]: probably it's not needed

Comment (Pawkkie) [reply docs/scope.md:9]: Insofar as the term "base" is used in the doc, the statements should hold true for both `master` and `upcoming`. I'm equally fine to leave it as is or modify the statement to "A .gba file built from an unmodified `master` or `upcoming` branch of `pokeemerald-expansion`. "

Comment (AsparagusEduardo) [reply docs/scope.md:9]: > Insofar as the term "base" is used in the doc, the statements should hold true for both `master` and `upcoming`. I'm equally fine to leave it as is or modify the statement to "A .gba file built from an unmodified `master` or `upcoming` branch of `pokeemerald-expansion`. "

This makes the most sense to me :)

File: docs/scope.md (hunk 10 above)
Comment (pkmnsnfrn) [reply docs/scope.md:45]: this implies that poryscript is in scope, whereas we explicitly voted that it wasn't

Comment (AsparagusEduardo) [reply docs/scope.md:45]: Then how about "external executables or build tools actively maintained by a different team"?

Comment (Pawkkie) [reply docs/scope.md:45]: I'm not sure how that's different than just a blanket "external tools are outside of scope", are you trying to leave an opening for something specific?

Comment (AsparagusEduardo) [reply docs/scope.md:45]: Mainly to allow us to have our own tools and not get lawyered by people saying that we technically have tools such as the teachable learnset helper

Comment (Pawkkie) [reply docs/scope.md:45]: Forgive me if I'm misunderstanding, but sure those are _internal_ tools?

Comment (AsparagusEduardo) [reply docs/scope.md:45]: > Forgive me if I'm misunderstanding, but surely those are _internal_ tools?

...you got me there 😂
Alright, the current description for this makes sense then xD

Comment (Pawkkie) [reply docs/scope.md:45]: Are we good for approval and merging then lol

**Review by AsparagusEduardo — APPROVED** [review]

File: docs/scope.md (hunk 10 above)
Comment (AlexOn1ine) [reply docs/scope.md:45]: what is an external tools and an internal? 

Can an external tool not become an internal tool?

Comment (AlexOn1ine) [reply docs/scope.md:45]: trainerproc was an external tool until it became an internal tool. Same with the teachable learnset tool

Comment (Pawkkie) [reply docs/scope.md:45]: Internal = expansion made for expansion use and thus expansion supported and included in our repo
External = supporting tool for decomp hacking that may or may not be expansion compatible but is maintained by others outside of our repo

> Can an external tool not become an internal tool?

Sure! But they are not in scope by default.

Comment (AlexOn1ine) [reply docs/scope.md:45]: I actual disagree here. Just because an we agreed Poryscript is out of scope doesn't mean other tools can't be in scope

Comment (Pawkkie) [reply docs/scope.md:45]: So you'd move it to "Discussion Required Categories" instead?

Comment (AlexOn1ine) [reply docs/scope.md:45]: Right, Idt a good tool should be blocked because we set a rule for it.

Comment (pkmnsnfrn) [reply docs/scope.md]: 9b06f7e4924604b8f69619009829ab0ba1dee437

if anybody disagrees, please propose an alternative

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Just realized that this should be targetted to master and a link to it be added to the website

**Pull request merged**